### PR TITLE
update test action to use CLI

### PR DESCRIPTION
**Pull request by samruddhikhandale**
Using `devcontainer templates apply` command to configure the default templates options.

Working demo on sample repo - https://github.com/samruddhikhandale/templates-testing-pipeline/actions/runs/3381285799/jobs/5615030578

Thanks @joshspicer for a super useful cmd ✨

### CODE REVIEW

**Review by joshspicer — APPROVED** [review]
😎

File: .github/workflows/test-pr.yaml [hunk 1]
```diff
@@ -67,35 +67,12 @@ jobs:
...
-                  exit 1
-                fi
-
-                echo "(!) Replacing '${option_key}' with '${option_value}'"
-                option_value_escaped=$(sed -e 's/[]\/$*.^[]/\\&/g' <<<"${option_value}")
-                find ./ -type f -print0 | xargs -0 sed -i "s/${option_key}/${option_value_escaped}/g"
-              done
-            fi
-          fi
+          devcontainer templates apply --workspace-folder "src/${{ matrix.templates }}" --template-id "ghcr.io/${{ github.repository }}/${{ matrix.templates }}:latest"
```
Comment (samruddhikhandale) [inline .github/workflows/test-pr.yaml:75]: @joshspicer  Is there a bug when we don't pass template args?
Looks like the CLI is returning incorrect list of changed files 🤔 

image

Comment (samruddhikhandale) [reply .github/workflows/test-pr.yaml:75]: It replaces correctly, the only issue with the returned `files[]`

Comment (samruddhikhandale) [reply .github/workflows/test-pr.yaml:75]: Oh wait. It also looks like the cmd is not just replacing the arguments, but it is also replacing the contents of the files (fetched from OCI artifact vs what is provided by the `workspace-folder`).

Hmm. I think that was how it was expected for `Add dev Container Config Cmd`. But in this case we can't use the cmd without tweaking or adding a new flag

**Pull request closed without merging**